app/src/main/java/emu/skyline/EmulationActivity.kt

```diff
@@ -314,6 +338,12 @@ class EmulationActivity : AppCompatActivity(), SurfaceHolder.Callback, View.OnTo
         }
     }
 
+    override fun onUserLeaveHint() {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.S) {
+            enterPictureInPictureMode(PictureInPictureParams.Builder().build())
+        }
```

Comment:
```suggestion
        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.S)
            enterPictureInPictureMode(PictureInPictureParams.Builder().build())
```

Reply:
> I wonder if this will prevent killing Skyline in lieu of a foreground notification.

To some extent. As long as the window stays on the screen, it's as if the app is running. If someone kills the PiP, then it's as if they exited the app. Not unlike a notification, but it won't survive sleep (as far as I know).

I should also see if there is a way to add a mute button or at least make one available in the window. I can think of a few cases that may be needed (ie a phone call).

Reply:
It might also be possible to add a pause button where you just kill the native surface from Java, that should keep Skyline resident in memory but let it remain pausable at will. While this is probably not the best way to pause things and we'll likely replace it in the future, it does work for everything so far afaik. You can do this by making `Surface?` a member of `EmulationActivity` where it can be assigned to in `surfaceCreated` and null'd in `surfaceDestroyed`, to pause you can simply call `setSurface(null)` and to resume you can call `setSurface(surface)`.